system/jlib/jptree.cpp

```diff
@@ -3049,15 +3049,23 @@ void PTree::deserializeFromStream(IBufferedSerialInputStream &src)
 {
     deserializeSelf(src);
 
+    NextByteStatus status{NextByteStatus::endOfStream};
     for (;;)
     {
-        if (isNextByteZero(src))
+        switch (status = isNextByteZero(src))
         {
-            src.skip(1); // Skip over null terminator.
-            break;
+            case NextByteStatus::nextByteisNonZero:
+            {
+                IPropertyTree *child = create(src);
+                addPropTree(child->queryName(), child);
+                break;
+            }
+            case NextByteStatus::nextByteisZero:
+                src.skip(1); // Skip over null terminator.
+                return;
+            default:
```

Comment:
better to explicitly make this case NextByteStatus::endOfStream, and a default that does throwUnexpected();